uvicorn/middleware/proxy_headers.py

```diff
@@ -59,8 +59,15 @@ async def __call__(
                 if b"x-forwarded-proto" in headers:
                     # Determine if the incoming request was http or https based on
                     # the X-Forwarded-Proto header.
-                    x_forwarded_proto = headers[b"x-forwarded-proto"].decode("latin1")
-                    scope["scheme"] = x_forwarded_proto.strip()
+                    x_forwarded_proto = (
+                        headers[b"x-forwarded-proto"].decode("latin1").strip()
+                    )
+                    if scope["type"] == "websocket":
+                        scope["scheme"] = (
```

Comment:
This line is missing a test coverage. Can you please add that?

Reply:
> I know there are no unit tests, because I don't know how to write, and help wanted.

Can you give me some examples of how to do a websocket test client? It seems that `httpx` does not support it.

Reply:
That issue is not related